### PR TITLE
[#55801] Add form validation in add project dialog for project attributes

### PR DESCRIPTION
**Pull request by judithroth**
OP#55801

### CODE REVIEW

File: app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+See COPYRIGHT and LICENSE files for more details.
+
+++#%>
+
+<%=
+  component_wrapper do
+    primer_form_with(
+      class: "op-new-project-mapping-form",
```
Comment (akabiru) [inline app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb:33]: 🟢 This class seems unused, can we delete it?

```suggestion
```

File: app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      model: @project_mapping,
+      url: link_admin_settings_project_custom_field_path(@project_custom_field),
+      data: { turbo: true },
+      method: :post
+    ) do |form|
+      concat(render(Primer::Alpha::Dialog::Body.new(
+        id: DIALOG_BODY_ID, test_selector: DIALOG_BODY_ID, aria: { label: title },
+        classes: "FormControl-horizontalGroup--sm-vertical",
```
Comment (akabiru) [inline app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb]: 🟠 Q: Do we need to define the vertical layout on mobile if we're using the vertical layout in both cases?

Seemed to have no added effect for me on mobile layout

Screenshot 2024-06-25 at 2 41 46 PM

File: frontend/src/global_styles/primer/_overrides.sass [hunk 3]
```diff
@@ -67,16 +67,9 @@ page-header
 
 // For some specific cases like progress form in small screens,
 // we need to change the direction of elements to column,
-// therefore elements are placed below each other.
-//
-// body needed for higher specificity compared to .FormControl-horizontalGroup--center-aligned
-body .FormControl-horizontalGroup--sm-vertical
+// therefore elements are placed below each other
+.FormControl-horizontalGroup--sm-vertical
```
Comment (akabiru) [inline frontend/src/global_styles/primer/_overrides.sass:71]: Happy to see this go back to "normal" 😄

**Review by akabiru — COMMENTED** [review]
Looks good to me, Judith! 👍🏾  I left some remarks tagged by the traffic light system (🟢 🟠 🔴 )

---

Some general Qs:

- [ ] 🟠 I noticed that we have a bit of a flicker now due to the split dialog render- I presume this is acceptable UX as the `respond_with_dialog` already existed?

  https://github.com/opf/openproject/assets/17295175/4c7dfdc3-56fb-457c-9593-2c65d9ce8d63


- [x] 🟢 I wonder if there's any further padding or restructuring we can do to the modal body to stop the "X" icon hover state touching the autocompleter input. Right now it's a bit of an eye sore 😬 

  
Screenshot 2024-06-25 at 2 36 40 PM

File: app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb (hunk 1 above)
Comment (judithroth) [reply app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb:33]: This class is adding the magnifier icon to the autocomplete dialog, see ` app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_component.sass:3` .
If we don't need the icon, I can remove it :slightly_smiling_face:

File: app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb (hunk 2 above)
Comment (judithroth) [reply app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb]: Good catch! We don't need it and I will remove it.

File: frontend/src/global_styles/primer/_overrides.sass (hunk 3 above)
Comment (judithroth) [reply frontend/src/global_styles/primer/_overrides.sass:71]: Me too :sweat_smile:

**Comment by judithroth** [comment]
> Looks good to me, Judith! 👍🏾 I left some remarks tagged by the traffic light system (🟢 🟠 🔴 )
> 
> Some general Qs:
> 
>     * [ ]  🟠 I noticed that we have a bit of a flicker now due to the split dialog render- I presume this is acceptable UX as the `respond_with_dialog` already existed?

I assume this is the case. After you mentioned it, I saw the flickering too (it only happens on the first time after the page is loaded). I used the pattern described in [the lookbook](https://qa.openproject-edge.com/lookbook/pages/patterns/dialogs) "Async dialogs as turbo streams". Actually I'm not sure how I could fix it.

>         
>           
>           
>       
>           flicker-add-projects-attributes.mp4
> 
>     * [ ]  🟢 I wonder if there's any further padding or restructuring we can do to the modal body to stop the "X" icon hover state touching the autocompleter input. Right now it's a bit of an eye sore 😬
> 
> 
> Screenshot 2024-06-25 at 2 36 40 PM

Well... one of my last code clean ups obviously removed the dialog header text. After readding the header title the eye sore is gone again and the icon hover and the input don't snuggle any more:

Screenshot from 2024-06-25 15-23-17

File: app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb (hunk 1 above)
Comment (akabiru) [reply app/components/settings/project_custom_fields/project_custom_field_mapping/new_project_mapping_form_component.html.erb:33]: Ah, my bad- I missed that!

**Review by akabiru — APPROVED** [review]
Thanks for addressing the feedback, looking good! :shipit:

**Comment by akabiru** [comment]
⚠️ @judithroth reminder to change the branch base to `dev`

**Comment by judithroth** [comment]
Force push / rebase because of changing the base branch - when I started working on this, I thought it was going into the release/14.2 branch, but it should go into dev.

**Pull request merged**